File: AkkaRemotingExample/RemoteNodeApp/src/main/scala/org/akka/essentials/remotenode/RemoteNodeApplication.scala
package org.akka.essentials.remotenode
import akka.kernel.Bootable
import akka.actor.ActorSystem
import akka.actor.Props
import com.typesafe.config.ConfigFactory

class RemoteNodeApplication extends Bootable {
  val system = ActorSystem("RemoteNodeApp", ConfigFactory
    .load().getConfig("RemoteSys"))

  def startup = {
    system.actorOf(Props[RemoteActor], name = "remoteActor")
  }

  def shutdown = {
    system.shutdown()
  }
}

File: AkkaRemotingExample/RemoteNodeApp/src/main/java/org/akka/essentials/remotenode/RemoteNodeApplication.java
package org.akka.essentials.remotenode;

import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.kernel.Bootable;

import com.typesafe.config.ConfigFactory;

public class RemoteNodeApplication implements Bootable {
	final ActorSystem system = ActorSystem.create("RemoteNodeApp", ConfigFactory
			.load().getConfig("RemoteSys"));

	public void shutdown() {
		system.shutdown();
	}

	public void startup() {
		system.actorOf(new Props(RemoteActor.class), "remoteActor");
	}
}
